x-pack/plugin/apm-integration/build.gradle

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+apply plugin: 'elasticsearch.internal-es-plugin'
+
+esplugin {
+  name 'x-pack-apm-integration'
+  description 'Provides APM integration for Elasticsearch'
+  classname 'org.elasticsearch.xpack.apm.APM'
+  extendedPlugins = ['x-pack-core']
```

Comment:
I tried moving it to `server` without a separate loader, and the agent code (not the server code) exploded all over the place with security manager exceptions. It's not obvious to be what is the problem, or why using another loader would help, since we don't directly use the agent jar, though I wonder if having it in the server classpath is enough to breaking things.

In the meantime, I've moved the APM code to be a regular module. Maybe this is something we can improve later.

Reply:
A regular module is fine for now. We can iterate later to load it differently.

Reply:
However, note that I don't think it would be possible to load apm early enough through a dynamic agent if it's in an ES module.

> I tried moving it to server without a separate loader

By loading directly from server I meant in a separate loader, similar to how we load x-content. This would mean the apm code would be under libs, need to be in a separate directory under libs in the distribution, and we would then create the loader on startup and reflectively add the agent.